app/src/main/java/com/ncnf/map/ui/MapFeedFragment.java

```diff
@@ -0,0 +1,73 @@
+package com.ncnf.map.ui;
+
+import android.view.View;
+import android.view.Window;
+import android.widget.Button;
+import android.widget.FrameLayout;
+
+import androidx.activity.OnBackPressedCallback;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.ncnf.R;
+import com.ncnf.feed.ui.FeedFragment;
+import com.ncnf.socialObject.SocialObject;
+import com.ncnf.socialObject.ui.SocialObjFragment;
+
+import java.util.List;
+
+public class MapFeedFragment extends FeedFragment {
+
+    private final Window globalWindow;
+    private final FragmentManager fragmentManager;
+
+    protected MapFeedFragment(List<SocialObject> list, Window globalWindow, FragmentManager fragmentManager){
+        super(list);
+
+        this.globalWindow = globalWindow;
+        this.fragmentManager = fragmentManager;
+    }
+
+    @Override
+    protected void onEventClick(SocialObject e){
+        Fragment fragment = new SocialObjFragment(e);
+
+        ConstraintLayout feedContainer = globalWindow.findViewById(R.id.map_feed_container);
+        FrameLayout feedFrame = globalWindow.findViewById(R.id.map_feed_fragment);
+        Button feedButton = globalWindow.findViewById(R.id.map_feed_button);
+
+        feedContainer.setBackgroundResource(R.drawable.main_background_gradient);
+        feedContainer.setVisibility(View.VISIBLE);
+        feedFrame.setVisibility(View.VISIBLE);
+        feedButton.setVisibility(View.VISIBLE);
+        feedButton.setText("Back to Feed");
+
+        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+        fragmentTransaction.replace(R.id.map_feed_fragment, fragment).addToBackStack(null).commit();
+
+        OnBackPressedCallback callback = new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                destroyChildFragment(fragmentManager, feedButton, feedContainer, this);
```

Comment:
I don't know, I'd be afraid to screw up the back navigation by doing other things than just pushing and popping on the stack. Plus I think it would create the issue of having 12 fragments on the stack if one person clicks on multiple markers. Which I'd guess is heavier than just recreating a fragment. 
We can definitely discuss it tho

Reply:
And include it in a later PR